client/my-sites/checkout/composite-checkout/lib/contact-validation.tsx

```diff
@@ -48,14 +48,21 @@ const getEmailTakenLoginRedirectMessage = (
 
 	// Users with a WP.com account should return to the checkout page
 	// once they are logged in to complete the process. The flow for them is
+
 	// checkout -> login -> checkout.
 	const currentURLQueryParameters = Object.fromEntries( new URL( href ).searchParams.entries() );
+
+	if ( currentURLQueryParameters.source ) {
```

Comment:
This is mostly to demonstrate the issue. There is likely a better fix elsewhere. I'm not sure what the `source` parameter is used for, but it's set on all product CTAs in cloud.jetpack.com/pricing.